src/bin/add/main.rs

```diff
@@ -95,9 +95,29 @@ fn print_msg(dep: &Dependency, section: &[String], optional: bool) -> Result<()>
     };
     write!(output, " {}", section)?;
     if let Some(f) = &dep.features {
-        writeln!(output, " with features: {:?}", f)?
+        if inline {
+            write!(output, " with features: {:?}", f)?;
+        } else {
+            writeln!(output, " with features: {:?}", f)?;
+        };
+    }
+    if !inline {
+        writeln!(output, ". ")?;
     } else {
-        writeln!(output)?
+        write!(output, ". ")?;
+    };
+
+    if !&dep.available_features.is_empty() {
+        if inline {
+            write!(output, " Optional features: [")?;
```

Comment:
This says "optional" which to me would imply ones not enabled by the user (either directly or through defaults)

Reply:
Fair. I think the easiest solution is just to change the wording, so I changed it to `Available`. I think filtering options that the user hasn't enabled is more in line with the validation of features which I wanted to keep in a separate PR, but that could be done if you think it's a good idea.